From 00ee1adfe1e34d43494227781f6662b0a21b7c4b Mon Sep 17 00:00:00 2001
From: Michael Jeanson <mjeanson@efficios.com>
Date: Tue, 21 Nov 2017 11:11:15 -0500
Subject: [PATCH] Fix: specify SONAME in python-lttngust LoadLibrary

When loading the python agent library with ctypes in the python
bindings, specify the SONAME. This will make sure we load the proper
library in the event of a SONAME bump and the bindings will work without
having to install the "dev" package which in most distros contains the
non-versionned ".so".

Signed-off-by: Michael Jeanson <mjeanson@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 python-lttngust/lttngust/loghandler.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python-lttngust/lttngust/loghandler.py b/python-lttngust/lttngust/loghandler.py
index e82cf5c5..6f144cac 100644
--- a/python-lttngust/lttngust/loghandler.py
+++ b/python-lttngust/lttngust/loghandler.py
@@ -22,7 +22,7 @@
 
 
 class _Handler(logging.Handler):
-    _LIB_NAME = 'liblttng-ust-python-agent.so'
+    _LIB_NAME = 'liblttng-ust-python-agent.so.0'
 
     def __init__(self):
         super(self.__class__, self).__init__(level=logging.NOTSET)
